### ISSUE TITLE
One file missing from coverage

### ISSUE THREAD

**Issue by crrobinson14** [istanbuljs/nyc#686]
## Expected Behavior

All files in a certain directory instrumented and reported on for coverage. Four files are here:

![image](https://user-images.githubusercontent.com/1675563/31186567-391f13e8-a8ec-11e7-97b7-0a76c64faca5.png)

The test suite includes four tests, each of which exercises one of these files. The code within the `createFlag.js` file is successfully executed:

![image](https://user-images.githubusercontent.com/1675563/31186570-3cfd8aee-a8ec-11e7-8129-5e8c9291c54d.png)

## Observed Behavior

The file `actions/flags/createFlag.js` is missing from the coverage output. All other files in the entire project (there are 30+ files in various folders) are properly instrumented and reported on:

![image](https://user-images.githubusercontent.com/1675563/31186560-3409f44a-a8ec-11e7-80f3-a10e28e8007d.png)

## Environment

https://gist.github.com/crrobinson14/b09ae1d65eade891cdd26c811e940dc9

```
  "scripts": {
        "test": "cross-env NODE_ENV=test node_modules/.bin/nyc --reporter=text-summary --reporter=html mocha"
   },
  "nyc": {
    "exclude": [
      "config/**/*.js",
      "bin/**/*.js",
      "test/**/*.js"
    ]
  },
```

This project is NodeJS-only and does not include Babel. There is no `.babelrc` file in the folder. There are no other exclusions specified anywhere. There are no Istanbul "ignore" comments in any of these files. I have tested using various combinations of "all" and "manually requiring the file" in the test.

I realize this is a mystery that may not be solved in a single response. What I'm asking for here is any ideas or guidance about how I can even begin to debug this issue. The test suite is running 100% properly. I have verified that the code within this file is actually being executed in various ways (e.g. console logging output from it). I can't for the life of me figure out why it's not included in the coverage report.

Where do I start? Is there some way to view verbose output from the instrumenter and reporter to determine where and why it might be getting excluded?

**Comment by crrobinson14** [comment]
For some reason I can't identify (`npm update` reported no modules to update, and on a whim I had tried a `rm -rf node_modules/.cache` to no effect), a complete removal of `node_modules` and `package-lock.json` followed by an `npm i` has resolved this issue.

**Comment by aneeshmg** [comment]
I have just encountered this problem, the solution that worked for @crrobinson14  does not work for me. I have a file whose coverage reports are not being shown by nyc. I even tried adding a nycrc with just the missing file in the "include" parameter and the result of that was 
----------|----------|----------|----------|----------|----------------|
File      |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
----------|----------|----------|----------|----------|----------------|
All files |  Unknown |  Unknown |  Unknown |  Unknown |                |
----------|----------|----------|----------|----------|----------------|
Am i missing anything?

**Comment by fdescamps** [comment]
I confirm the bug and how to correct it by @crrobinson14  ;)

Thanks !

**Comment by thaddeusreid** [comment]
I am experiencing the same problem as described by @aneeshmg 

Two or three directories that match the glob pattern in my `include` values do not appear in any of the coverage reports. Explicitly targeting the missing directories returns an empty report.

```
----------|----------|----------|----------|----------|-------------------|
File      |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
----------|----------|----------|----------|----------|-------------------|
All files |        0 |        0 |        0 |        0 |                   |
----------|----------|----------|----------|----------|-------------------|
```

Might be worth opening a new Issue to look into this bad behavior.

**Comment by tbell511** [comment]
I am having the same problem. I deleted node_modules and my lock file, and reinstalled all dependencies but no luck.

**Comment by crrobinson14** [comment]
@tbell511 Did you see the comment the core team made in #980? They're hoping somebody can provide a repo with sample code that reproduces the issue. The trouble is a lot of us are seeing this, but having to move on with workarounds and not always able to provide an example. If you have one it would be great!

**Comment by tbell511** [comment]
@crrobinson14 thanks for the reply.  Yes, I did see the comment but I am in the same boat as most.  My repo is private.  I would be willing and happy to get on a call though with someone from the core team to investigate my repo though.  Maybe a screen share would help them out. 

That being said, last night while I was trying to fix the issue I made it worse :( After I deleted my node_modules folder, and lock file, nyc is not covering any of my files.  Nothing changed between the deleting of the node_modules and lock file.  Even if I specify a specific file with include, it doesn't work.  

Any ideas on what could have caused nyc to stop covering all files?

**Comment by brandonpas** [comment]
I'm not sure if this will help anyone with their specific situation, but I was encountering the same issue described. I enabled strict mode ('use strict') on my files and found my unit tests started failing due a few variables not being initialized with "var". Upon resolving these issues, the missing files were included in the coverage report.

**Comment by mayo84** [comment]
I'm getting this weird experience too. You can use this project: https://github.com/Alberto85ariel13/vts where i'm a collaborator. I'm using nyc with mocha and ts-node/register config for tests.
All mocha tests are passing but nyc is working completely crazy in that some times it convers all files in /src directory but other times it only cover 10 files.
To reproduce this behavior, just keep doing npm run test many times and you will eventually run into the problem.
Don't know how to solve this and becouse i have integration with coveralls.io, some times my coverage data only covers 10 files when my CI runs the coverage workflow.

**Comment by tbell511** [comment]
If this gets fixed I will be super happy. I have had to stop using NYC because of this very issue.

**Comment by markf-hubdoc-com** [comment]
> For some reason I can't identify (`npm update` reported no modules to update, and on a whim I had tried a `rm -rf node_modules/.cache` to no effect), a complete removal of `node_modules` and `package-lock.json` followed by an `npm i` has resolved this issue.

For me just the `rm -rf node_modules` followed by `npm i` resolved the issue. Interesting. Thanks for the tip, @crrobinson14!

**Comment by crrobinson14** [comment]
My pleasure, @markf-hubdoc-com . For what it's worth, I haven't had this issue recur in a long time. I switched to `yarn` as a package manager last year. Possibly that had something to do with it?

**Comment by markf-hubdoc-com** [comment]
> My pleasure, @markf-hubdoc-com . For what it's worth, I haven't had this issue recur in a long time. I switched to `yarn` as a package manager last year. Possibly that had something to do with it?

Good to know, @crrobinson14. Still on `npm` over here - could be due to some code running on older npm (v5.6.0) with Node.js 8. Will keep your comment in mind as we upgrade to Node.js 10 and 12 with the corresponding npm upgrades. Thanks!

**Comment by TimBailey-pnk** [comment]
I've discovered that when I see this issue, i can resolve it by updating to the latest "sub-dependecies" by running `yarn upgrade` (i can do this as I run fixed versions for my dependencies)

**Comment by thisisashwani** [comment]
I faced this issue too. Turns out `nyc` ignores files which has issues that, in general, won't throw an error either. In my case, i had two functions with same name in my file. This was by mistake, and shouldn't be there. But since JS doesn't throw error for this, I had no idea till I started to check after nyc coverage was missing for this file.

Good option - Use a default linting guide like the one from Airbnb - https://github.com/airbnb/javascript

**Comment by ashwani-pandey** [comment]
I faced this issue too. Turns out `nyc` ignores files which has issues that, in general, won't throw an error either. In my case, I had two functions with same name in my file. This was by mistake, and shouldn't be there. But since JS doesn't throw error for this, I had no idea till I started to check after nyc coverage was missing for this file.

Good option - Use a default linting guide like the one from Airbnb - https://github.com/airbnb/javascript

**Comment by thimpat** [comment]
Quick note, and similar to @ashwani. After using arguments as a variable name (with const, not let), nyc would break and not render any coverage (on that file only, but the rest would be fine).
```
function()
{
  if (something)
  {
     const arguments = ...
  }
}
```
It's not something that is raised by linters or Js.

**Comment by shukriadams** [comment]
I got this error as well, @ashwani-pandey's suggestion pointed me in the right direction. One of my files wasn't getting picked up despite having 100% coverage and passing tests. Turns out I had a variable called "package", which I suppose is legal but ill-advised. If you have a file that refuses to show up in your coverage list, comment all code out and rerun cover, if the file shows up, you likely have a soft error somewhere.

**Comment by tamilvendhank** [comment]
I ran into this issue today. Deleting `node_modules` and `package-lock.json` did not solved the issue in my case. I undid my recent changes & could see the same file appear in the reporting again. So, I checked the recently added method's content. There was a variable with the name `interface`. It was causing the issue. So, I renamed it to `_interface`(was checking) & it solved the issue. So, I guess, files missing from report has also to do something with some kind of reserved names.
